ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKTable.java

```diff
@@ -79,9 +81,9 @@ public SchemaKTable into(
     if (isWindowed) {
       final Serde<Windowed<String>> windowedSerde
               = WindowedSerdes.timeWindowedSerdeFrom(String.class);
-      ktable.toStream()
+      ((KTable<Windowed<String>, GenericRow>)ktable).toStream()
```

Comment:
You could use two constructors and two private inner class implementations to handle windowed and non-windowed KTables, with the correct generics, e.g. one constructor that took a `KTable<Windowed<String>, GenericRow>` and one that took a `KTable<String, GenericRow>`.

this would make the whole thing more type safe and remove the needs for the casts.

Reply:
That won't work because both methods would have the same erasure. We could use static factory functions, but even then they wouldn't actually check anything since we're passing around raw KTables/KStreams/KGroupedStreams/KGroupedTables in most places. In the interest of keeping this PR efficient how about I file an issue for actually using type parameters with these generics and enforcing type safety?